File: backend/src/schemas/bootstrap.py
# backend/src/schemas/bootstrap.py
from typing import Optional, List, Literal
from pydantic import BaseModel, Field

from .telegram import TelegramUserSchema  # уже есть

# Короткие карточки сущностей для фронта
class BotBrief(BaseModel):
    id: int
    username: Optional[str] = None
    name: Optional[str] = None
    telegram_id: Optional[int] = None
    is_active: bool = True
    bot_url: Optional[str] = None

class GroupBrief(BaseModel):
    id: int
    telegram_id: int
    name: str
    type: str
    is_active: bool = True
    photo: Optional[str] = None

class CommunityBrief(BaseModel):
    id: int
    table_key: str
    domain: str

class CommunityAggregate(BaseModel):
    community: CommunityBrief
    role: Optional[Literal["owner", "admin", "moderator"]] = None
    bot: Optional[BotBrief] = None
    main_group: Optional[GroupBrief] = None
    additional_group: Optional[GroupBrief] = None

class BootstrapResponse(BaseModel):
    user: TelegramUserSchema
    communities: List[CommunityAggregate] = Field(default_factory=list)

File: frontend/src/types/bootstrap.ts
// types/bootstrap.ts
export type BotBrief = {
  id: number;
  username?: string | null;
  name?: string | null;
  telegram_id?: number | null;
  is_active: boolean;
  bot_url?: string | null;
};

export type GroupBrief = {
  id: number;
  telegram_id: number;
  name: string;
  type: string;
  is_active: boolean;
  photo?: string | null;
};

export type CommunityBrief = {
  id: number;
  table_key: string;
  domain: string;
};

export type CommunityAggregate = {
  community: CommunityBrief;
  role?: "owner" | "admin" | "moderator" | null;
  bot?: BotBrief | null;
  main_group?: GroupBrief | null;
  additional_group?: GroupBrief | null;
};

export type TelegramUserSchema = {
  id: number;
  first_name: string;
  last_name?: string | null;
  username?: string | null;
  language_code?: string | null;
  is_premium?: boolean;
  photo_url?: string | null;
};

export type BootstrapResponse = {
  user: TelegramUserSchema;
  communities: CommunityAggregate[];
};
